ibc-core/ics02-client/types/src/msgs/mod.rs

```diff
@@ -1,4 +1,6 @@
 //! Defines the client message types that are sent to the chain by the relayer.
+#![allow(deprecated)]
```

Comment:
I ended up having to add this `allow(deprecated)` attribute at the top module level, as no matter how I structured `#[allow(deprecated)]` attributes on the `ClientMsg` enum, the compiler always threw an error regarding the `Misbehaviour(MsgSubmitMisbehaviour)` variant. Adding the `[allow(deprecated)]` attribute at the top of the `mod.rs` file was the only way that silenced that error. Not sure this is desirable though.

Reply:
It might not be super efficient, but that's okay imo. Let's have our handlers support both `MsgUpdateClient` and `MsgSubmitMisbehaviour` for a few versions. We can consider phasing it out completely before v1.